cybersecurity/bash-system-monitor/devashmit-day38/monitor.sh

```
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+# Bash System Monitoring Tool — Day 38 Cybersecurity | Author: devashmit
+CPU_WARN=70;CPU_CRIT=90;RAM_WARN=75;RAM_CRIT=90;DISK_WARN=80;DISK_CRIT=95
+LOG="/var/log/sysmonitor.log";TS=$(date '+%Y-%m-%d %H:%M:%S')
+RED='\033[0;31m';YEL='\033[0;33m';GRN='\033[0;32m';BOLD='\033[1m';RST='\033[0m'
+colorize(){local v=$1 w=$2 c=$3;if(( $(echo "$v >= $c"|bc -l) ));then echo -e "${RED}${v}%${RST}";elif(( $(echo "$v >= $w"|bc -l) ));then echo -e "${YEL}${v}%${RST}";else echo -e "${GRN}${v}%${RST}";fi;}
+log(){echo -e "$1";echo "[$TS] $(echo "$1"|sed 's/\x1b\[[0-9;]*m//g')" >> "$LOG" 2>/dev/null||true;}
+get_cpu(){local idle;idle=$(top -bn1|grep "Cpu(s)"|awk '{print $8}'|tr -d '%,');[[ -z "$idle" ]]&&idle=0;printf "%.1f" "$(echo "100 - $idle"|bc -l)";}
```

Shellcheck (0.11.0)

[error] You need a space after the '{'.

(SC1054)